loadtest/user/userentity/websocket.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package userentity
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+const (
+	// Same as the webapp settings.
+	minWebsocketReconnectDuration = 3 * time.Second
+	maxWebsocketReconnectDuration = 5 * time.Minute
+	maxWebsocketFails             = 7
+)
+
+// listen starts to listen for messages on various channels.
+// It will keep reconnecting if the connection closes.
+// Only on calling Disconnect explicitly, it will return.
+func (ue *UserEntity) listen(errChan chan error) {
+	connectionFailCount := 0
+	for {
+		client, err := model.NewWebSocketClient4(ue.config.WebSocketURL, ue.client.AuthToken)
```

Comment:
Yes, sorry about that. We discussed offline on this. Using the `Connect` method to reconnect works, but it creates a nasty race condition when the user is executing client.SendMessage. This creates a race on the connection because at the same time, you are trying to replace the connection object and write to it.
Creating a new connection from scratch avoids this because it is an entirely new object. And the user will try to write to the old connection, which will fail, but there will be no races.